nuget/helpers/lib/NuGetUpdater/NuGetUpdater.Core/Utilities/DependencyConflictResolver.cs

```diff
@@ -0,0 +1,680 @@
+    private readonly Dictionary<PackageToUpdate, HashSet<PackageToUpdate>> packageDependencies = new Dictionary<PackageToUpdate, HashSet<PackageToUpdate>>();
+    private readonly Dictionary<PackageToUpdate, HashSet<PackageToUpdate>> reverseDependencies = new Dictionary<PackageToUpdate, HashSet<PackageToUpdate>>();
+
+    // Path of the repository
+    private readonly string repoRoot;
+
+    // Path to the project within the repository
+    private readonly string projectPath;
+
+    public PackageManager(string repoRoot, string projectPath)
+    {
+        this.repoRoot = repoRoot;
+        this.projectPath = projectPath;
+    }
+
+    // Method alterted from VersionFinder.cs to find the metadata of a given package
+    private async Task<IPackageSearchMetadata?> FindPackageMetadataAsync(PackageIdentity packageIdentity, CancellationToken cancellationToken)
+    {
+        string? currentDirectory = Path.GetDirectoryName(projectPath);
+        string CurrentDirectory = currentDirectory ?? Environment.CurrentDirectory;
+        SourceCacheContext SourceCacheContext = new SourceCacheContext();
+        PackageDownloadContext PackageDownloadContext = new PackageDownloadContext(SourceCacheContext);
+        ILogger Logger = NullLogger.Instance;
+
+        IMachineWideSettings MachineWideSettings = new NuGet.CommandLine.CommandLineMachineWideSettings();
+        ISettings Settings = NuGet.Configuration.Settings.LoadDefaultSettings(
+            CurrentDirectory,
+            configFileName: null,
+            MachineWideSettings);
+
+        var globalPackagesFolder = SettingsUtility.GetGlobalPackagesFolder(Settings);
+        var sourceMapping = PackageSourceMapping.GetPackageSourceMapping(Settings);
+        var packageSources = sourceMapping.GetConfiguredPackageSources(packageIdentity.Id).ToHashSet();
+        var sourceProvider = new PackageSourceProvider(Settings);
+
+        ImmutableArray<PackageSource> PackageSources = sourceProvider.LoadPackageSources()
+             .Where(p => p.IsEnabled)
+             .ToImmutableArray();
+
+        var sources = packageSources.Count == 0
+            ? PackageSources
+            : PackageSources
+                .Where(p => packageSources.Contains(p.Name))
+                .ToImmutableArray();
+
+        var message = new StringBuilder();
+        message.AppendLine($"finding info url for {packageIdentity}, using package sources: {string.Join(", ", sources.Select(s => s.Name))}");
+
+        foreach (var source in sources)
+        {
+            message.AppendLine($"  checking {source.Name}");
+            var sourceRepository = Repository.Factory.GetCoreV3(source);
+            var feed = await sourceRepository.GetResourceAsync<MetadataResource>(cancellationToken);
+            if (feed is null)
+            {
+                message.AppendLine($"    feed for {source.Name} was null");
+                continue;
+            }
+
+            var existsInFeed = await feed.Exists(
```

Comment:
This call can throw if it's a private package source with bad authentication.  Wrap this section like is done [here](https://github.com/dependabot/dependabot-core/blob/4aac28c263db6b55c2c4ff326a57b49de034270e/nuget/helpers/lib/NuGetUpdater/NuGetUpdater.Core/Analyze/NuGetContext.cs#L107-L126).